[28기 이가윤] 어드민 페이지 구현

Gayun00 opened the pull request:
[Add] 이미지 업로드 값 인코딩 기능 구현

[Refactor] 리팩토링 중

[Add] 가격 변경, 호텔 삭제 기능 추가

## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 어드민 페이지 구현

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)

<br />

## :: 기타 질문 및 특이 사항
- 주석 처리한 부분은 백엔드 통신을 위한 코드입니다.

bonbon0202 commented on src/pages/Admin/Admin.js:
```diff
@@ -0,0 +1,187 @@
+    const uploadedFile = e.target.files[0];
+    reader.readAsDataURL(uploadedFile);
+    reader.onload = function (e) {
+      const encodedImg = e.target.result;
```
해당 값 사용해서 파일 업로드 했을 때 미리보기를 해주어도 좋을 것 같아요-!

Gayun00 replied:
네 한번 해볼게요!

bonbon0202 commented on src/pages/Admin/Admin.js:
```diff
@@ -0,0 +1,187 @@
+function Admin() {
+  const [hotelList, setHotelList] = useState([]);
+  const [price, setPrice] = useState(0);
+  const reader = new FileReader();
```
fileReader 잘 사용하셨네요! 굿굿

bonbon0202 commented on src/pages/Admin/Admin.js:
```diff
@@ -0,0 +1,187 @@
+  }, []);
+
+  function loadAdminData() {
+    fetch(`/data/admin${adminId}.json`)
```
유저가 기존에 등록한 호텔들이 나오는 데이터죠?!

Gayun00 replied:
넵 맞습니다...!

bonbon0202 commented on src/pages/Admin/Admin.js:
```diff
@@ -0,0 +1,187 @@
+    };
+  }
+
+  function updatePrice(e, id) {
```
id 값을 매개변수로 작성하신 이유가 있으실까요?!

Gayun00 replied:
mapping 해서 호텔들을 보여주어서, 이때 id값을 매개변수로 전달한 다음 받은 id값을 바로 요청하려고 합니다

bonbon0202 commented on src/pages/Admin/Admin.js:
```diff
@@ -0,0 +1,187 @@
+  }
+`;
+
+const UploadButton = styled.div`
+  //
+`;
+
+const TextContainer = styled.div`
+  //
+`;
+
+const Name = styled.h3`
+  //
+`;
+
+const Price = styled.p``;
+
+const PriceInputWrapper = styled.span``;
+
+const PriceInput = styled.input``;
+
+const PriceButton = styled.button``;
```
추후에 css가 추가되는 부분일까요?! css 속성이 불필요하다면 컴포넌트 형식이 아닌 요소로 작성해주셔도 됩니다!

Gayun00 replied:
넵 추후에 css 수정예정이라 비워뒀스니다!

bonbon0202 commented on src/pages/Admin/Admin.js:
```diff
@@ -0,0 +1,187 @@
+  }
+
+  async function deleteHotel(hotelId) {
+    // await fetch(`url/admins/${adminId}/?stay_id=${hotelId}`, {
+    //   method: 'PATCH',
+    //   headers: {
+    //     //
+    //   },
+    //   body: {
+    //     hotelId: hotelId,
+    //   },
+    // });
+    //reLoadHotelList();
```
해당 로직은 금일 보완이 될 내용이지요?!

Gayun00 replied:
넵 PR템플릿에 써뒀는데, 통신위해서 작성해둔 함수입니다! 오늘 보완하려고 합니다

Gayun00 committed:
[Fix] recoil 재설치

[Add] 어드민 페이지 레이아웃 추가

[Add] 이미지 업로드 값 인코딩 기능 구현

[Refactor] 리팩토링 중

[Add] 가격 변경, 호텔 삭제 기능 추가

[Fix]깃 충돌 수정

[Fix] recoil 재설치
